ydb/public/lib/ydb_cli/import/import.cpp

```diff
@@ -378,28 +432,51 @@ TStatus TImportFileClient::UpsertCsv(IInputStream& input, const TString& dbPath,
         splitter.ConsumeLine();
     }
 
-    std::vector<TAsyncStatus> inFlightRequests;
+    THolder<IThreadPool> pool = CreateThreadPool(settings.Threads_);
 
-    ui32 idx = settings.SkipRows_;
-    ui64 readSize = 0;
+    ui32 row = settings.SkipRows_;
     ui64 nextBorder = VerboseModeReadSize;
+    ui64 batchBytes = 0;
+    ui64 readBytes = 0;
+
+    TString line;
+    std::vector<TAsyncStatus> inFlightRequests;
+
     while (TString line = splitter.ConsumeLine()) {
+        ++row;
+        readBytes += line.Size();
+        batchBytes += line.Size();
         buffer += line;
         buffer += '\n';
-        readSize += line.size();
-        ++idx;
-        if (readSize >= nextBorder && RetrySettings.Verbose_) {
+
+        if (readBytes >= nextBorder && RetrySettings.Verbose_) {
             nextBorder += VerboseModeReadSize;
-            Cerr << "Processed " << 1.0 * readSize / (1 << 20) << "Mb and " << idx << " records" << Endl;
+            Cerr << "Processed " << 1.0 * readBytes / (1 << 20) << "Mb and " << row << " records" << Endl;
         }
-        if (buffer.Size() >= settings.BytesPerRequest_) {
-            auto status = WaitForQueue(inFlightGetter.GetCurrentMaxInflight(), inFlightRequests);
-            if (!status.IsSuccess()) {
-                return status;
-            }
 
-            inFlightRequests.push_back(UpsertCsvBuffer(dbPath, buffer));
-            buffer = headerRow;
+        if (batchBytes < settings.BytesPerRequest_) {
+            continue;
+        }
+
+        if (inputSizeHint && progressCallback)
+        {
+            double progress = static_cast<double>(readBytes) / *inputSizeHint;
+            progressCallback(static_cast<size_t>(progress * 100));
```

Comment:
Could we remove magic const 100 from here? I.e. we could call
progressCallback(readBytes, inputSizeHint)